aten/src/ATen/core/jit_type.h

```diff
@@ -1541,6 +1541,8 @@ TORCH_API std::ostream& operator<<(std::ostream& os, const Stride& s);
 // what is the type, ignoring extra size/shape information?
 // e.g. Tensor(2x3) -> Dynamic, and Tuple(Tensor(2x3),...) -> Tuple(Dynamic,...)
 
+// xxx: be careful with calls because this can be very slow. If calling this on a graph
+// use `EraseShapeInformation` in shape_analysis.h
 inline TypePtr unshapedType(const TypePtr& type) {
```

Comment:
should we make `unshapedType` return a singleton?

Reply:
wdym ?

Reply:
if you mean persistent caching, maybe, but that's a different PR. we would need to store weak ptrs and have all cache lock and such, i would wait till it's actually needed and/or fix callsites instead.